vault/resource_generic_secret.go

```diff
@@ -121,13 +137,23 @@ func genericSecretResourceWrite(d *schema.ResourceData, meta interface{}) error
 	}
 
 	log.Printf("[DEBUG] Writing generic Vault secret to %s", path)
-	_, err = client.Logical().Write(path, data)
+	secret, err := client.Logical().Write(path, data)
 	if err != nil {
 		return fmt.Errorf("error writing to Vault: %s", err)
 	}
 
 	d.SetId(originalPath)
 
+	if secret != nil {
+		jsonData, err := json.Marshal(secret.Data)
+		if err != nil {
+			return fmt.Errorf("error marshaling JSON for %q: %s", path, err)
+		}
+		d.Set("write_data_json", string(jsonData))
+	} else {
+		d.Set("write_data_json", "null")
```

Comment:
Is this line needed?

Reply:
Some write operations don't return anything, and I wanted to make sure that a reference to `write_data_json` didn't fail in that case. Basically I was trying to follow the pattern I observed that all things that could be read from the resource were explicitly initialized. Please let me know if I have misunderstood how this is supposed to work. I can probably produce an example that would error out if this line were absent. Basically since terraform doesn't have any way to ask whether a particular field is there, it's easier to work with if the same fields are always there but just have explicitly null values. Lazy evaluation of conditionals in 0.12 may make this less problematic.